addons/crm/models/crm_lead.py

```diff
@@ -2196,7 +2196,7 @@ def _pls_get_naive_bayes_probabilities(self, batch_mode=False):
         for team_id in result:
             result[team_id]['team_won'], \
             result[team_id]['team_lost'], \
-            result[team_id]['team_total'] = self._pls_get_won_lost_total_count(result[team_id])
+            result[team_id]['team_total'] = self.with_context(pls_team_id=team_id)._pls_get_won_lost_total_count(result[team_id])
```

Comment:
Hi @pca-Odoo ,

While more concise, we tend to avoid using context when we can. In this case the diff would not be much larger (even though there may be a bit of duplication) if we go with something like:
```python
team_won, team_lost, team_total = self._pls_get_won_lost_total_count(result[team_id])
if team_won == 0 and team_lost == 0 and team_total == 0 and team_id and team_id > 0:
    first_stage_id = self.env['crm.stage'].search([('team_id', '=', team_id)], order='sequence, id', limit=1)
    if str(first_stage_id.id) in team_results.get('stage_id', []):
      stage_result = team_results['stage_id'][str(first_stage_id.id)]
      team_won, team_lost, team_total = stage_result['won'], stage_result['lost'], stage_result['won'] + stage_result['lost']
```

Reply:
And then of course:
```python3
result[team_id]['team_won'], \
result[team_id]['team_lost'], \
result[team_id]['team_total'] = t_won, t_lost, t_total
```
Actually, let us rename to `t_won` `t_lost` `t_total` to avoid side effects with existing variables below